management/authenticators.go

```diff
@@ -0,0 +1,329 @@
+package ziti
+
+import (
+	"context"
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/openziti/edge-api/rest_management_api_client/authentication"
+	"github.com/openziti/edge-api/rest_model"
+	"github.com/openziti/identity"
+	"github.com/openziti/sdk-golang/ziti"
+	"github.com/openziti/sdk-golang/ziti/sdkinfo"
+	"net/url"
+)
+
+type ApiAuthenticator interface {
+	Authenticate(clientApiUrl *url.URL) (*AuthenticatedApiClient, *rest_model.CurrentAPISessionDetail, error)
+	AuthenticateWithClient(apiClient *ApiClient) (*rest_model.CurrentAPISessionDetail, error)
+	NewApiClient(clientApiUrl *url.URL) *ApiClient
+	SetInfo(envInfo *rest_model.EnvInfo, sdkInfo *rest_model.SdkInfo)
+	SetConfigTypes(configTypes []string)
+	GetBase() *BaseAuthenticator
+}
+
+type IdentityProvider interface {
```

Comment:
same redefinition here. i expect this is just an oversight? i don't want to make that same comment but there are __a lot__ of functions/interfaces/etc that seem duplicative?

Reply:
De-duping this one is possible.